root/static/scripts/common/i18n/expand2react.js

```diff
@@ -146,14 +146,17 @@ function pushChild<T>(
   if (lastIndex >= 0 &&
       typeof match === 'string' &&
       typeof children[lastIndex] === 'string') {
+    // $FlowIssue[incompatible-type]
     children[lastIndex] += match;
   } else {
     children.push(match);
   }
   return children;
 }
 
-function concatArrayMatch<T>(
+type MatchUpperBoundT = StrOrNum | {...};
```

Comment:
I'm not sure I follow what this one means.

Reply:
Added a comment here -- does that help some? `{...}` just matches any object.

Besides outputting React elements, another case where we parse a string into objects is HTML attributes:

https://github.com/metabrainz/musicbrainz-server/blob/0550d68/root/static/scripts/common/i18n/expand2react.js#L279-L282

(Then we pass that object as the second argument to `React.createElement`.)